### ISSUE TITLE
LightGBMTuner

### ISSUE THREAD

**Issue by BasLaa** [optuna/optuna#3219]
I have been trying to use the LightGBMTuner, but I am having two issues with it:

- I do not understand how to define the objective as you would do regularly in a Study. In the example on Medium, no explicit objective is defined either. Would it be feval in lgb.train() which is used for this? Or is the evaluation simply always set to be the standard objective function?
- How would you do cross-validation with LightGBMTuner. Again, In the example on Medium, a simple split is used and one part is used as validation set. I thought about just looping through folds of the full dataset and setting each as validation set in turn, but that would not make sense either, as the Tuner would find the best parameters for set 1, then set 2 etc, instead of averaging the results on each fold for some combination of hyperparameters.

## Information for reproducibility (optional)

```python
train_X, test_X, train_y, test_y = train_test_split(df_train[features], df_train[TARGET_COL], test_size=0.25) 

train_data = lgbm.Dataset(train_X, label=train_y)
eval_data = lgbm.Dataset(test_X, label=test_y)

param_grid = {
        # "device_type": trial.suggest_categorical("device_type", ['gpu']),
        # "objective": trial.suggest_categorical("objective", ['regression', 'rank_xendcg'])
        "n_estimators": 1500,
        "learning_rate": 0.04,
        "num_leaves": 32,
        "max_depth": 7,
        "objective": "regression",
        # "metric": "custom",
        "verbosity": 1,
    }


best = lightgbm_tuner.train(param_grid, 
                            train_data,
                            valid_sets=eval_data,
                            )
```

**Comment by nzw0301** [comment]
1. Defining the objective is not necessary for `LightGBMTuner`. `LightGBMTuner` wraps such logic as in the source code: https://github.com/optuna/optuna/blob/2c44c1a405ba059efd53f4b9c8e849d20fb95c0a/optuna/integration/_lightgbm_tuner/optimize.py#L633 
2. duplicated with https://github.com/optuna/optuna/issues/3203

**Comment by BasLaa** [comment]
Thank you for the swift response, 2. is clear to me. 

For 1, I'm not sure yet. If I want to define my own evaluation function for the cross validation happening inside LightGBMTunerCV, where would I define and enter this evaluation function? Would that be feval?

**Comment by nzw0301** [comment]
I'm not sure that the following answer, but maybe like this?

```python
def feval(preds, train_data):
    # custom metric
    # as a stupid example, return a random value
    custom_metric = np.random.rand()

    is_higher_better = True
    return "own_metric", custom_metric, is_higher_better

params = {
    "objective": "binary",
    "metric": "own_metric",
    "verbosity": -1,
    "boosting_type": "gbdt",
}

model = lgb.train(
    params,
    dtrain,
    valid_sets=[dtrain, dval],
    verbose_eval=-1,
    feval=feval,
)
```

Define the `feval` by following LightGBM documentation, for example, https://lightgbm.readthedocs.io/en/latest/pythonapi/lightgbm.train.html and set `metric` as the first returned value of `feval`, in the example above, `own_metric`
